src/pages/index.js

```diff
@@ -59,54 +32,126 @@ const config = {
   inputErrorClass: "modal__input_type_error",
   errorClass: "modal__error_visible",
 };
+const profileButton = document.querySelector(".profile__button");
+
+const profileAvatarModal = document.querySelector("#profile-avatar-modal");
+const profileImage = document.querySelector(".profile__image");
+const profileAvatarModalForm = profileAvatarModal.querySelector(".modal__form");
+
+const userInfo = new UserInfo(".profile__title", ".profile__description");
+const cardSection = new Section(renderCard, cardListEl);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f1dab6ed-e5ba-44ef-bef1-7369d7e9bf0d",
+    "Content-Type": "application/json",
+  },
+});
+
+api.getUserInfo().then((res) => {
+  userInfo.setUserInfo(res);
+});
+api.getInitialCards().then((res) => {
+  cardSection.renderItems(res);
+});
+api.getUserInfo().then((res) => {
+  profileImage.src = res.avatar;
+});
+
+const cardDeleteModal = new PopupQuestion("#delete-card-modal");
 
 const editFormValidator = new FormValidator(config, profileEditForm);
 editFormValidator.enableValidation();
 const addFormValidator = new FormValidator(config, addCardFormElement);
 addFormValidator.enableValidation();
+const profileAvatarFormValidator = new FormValidator(
+  config,
+  profileAvatarModalForm
+);
 
+profileAvatarFormValidator.enableValidation();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link, submitButton) => {
+    api.addCard(link, submitButton).then((res) => {
+      cardSection.addItem(res);
+    });
+    addFormValidator.toggleButtonState();
+    cardPopup.close();
```

Comment:
```js

    addFormValidator.toggleButtonState();
    cardPopup.close();
```

This should be done in `then`


You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


And `.toggleButtonState()` also


**Please, correct it everywhere**